modules/gridss/1.0/gridss.smk

```diff
@@ -0,0 +1,501 @@
+        opts = CFG["options"]["gridss"], 
+        steps = "assemble,call", 
+        jvmheap = lambda wildcards, resources: int(resources.mem_mb * 0.8) 
+    conda:
+        CFG["conda_envs"]["gridss"]
+    threads:
+        CFG["threads"]["gridss"]
+    resources:
+        **CFG["resources"]["gridss"]
+    priority: 2 
+    group: "enormous_bam"
+    shell:
+        op.as_one_line("""
+        gridss
+        --reference {input.fasta}
+        --output {output.vcf}
+        --workingdir `dirname {output.vcf}`
+        --assembly {output.assembly}
+        --blacklist {input.blacklist}
+        --repeatmaskerbed {input.repeatmasker}
+        --threads {threads}
+        --jvmheap {params.jvmheap}m
+        --labels "{wildcards.normal_id},{wildcards.tumour_id}"
+        --steps {params.steps}
+        {params.opts}
+        {input.normal_bam} 
+        {input.tumour_bam} 
+        2>&1 | tee -a {log}
+        """)
+   
+# Run GRIDSS in unpaired mode
+rule _gridss_unpaired:
+    input:
+        tumour_bam = CFG["dirs"]["inputs"] + "bam/{seq_type}--{genome_build}/{tumour_id}.bam",
+        tumour_preproc = CFG["dirs"]["gridss"] + "{seq_type}--{genome_build}/{tumour_id}--None--no_normal/{tumour_id}.bam.gridss.working",
+        t_workdir = CFG["dirs"]["preprocess"] + "{seq_type}--{genome_build}/{tumour_id}.bam.gridss.working", # Placeholder to prevent premature deletion of the temp workdir 
+        fasta = str(rules._gridss_input_references.output.genome_fa),
+        fasta_img = str(rules._gridss_setup_references.output.genome_img), 
+        blacklist = reference_files("genomes/{genome_build}/encode/encode-blacklist.{genome_build}.bed"), 
+        repeatmasker = reference_files("genomes/{genome_build}/repeatmasker/repeatmasker.{genome_build}.bed")
+    output:
+        vcf = CFG["dirs"]["gridss"] + "{seq_type}--{genome_build}/{tumour_id}--None--no_normal/gridss_raw.vcf.gz",
+        assembly = CFG["dirs"]["gridss"] + "{seq_type}--{genome_build}/{tumour_id}--None--no_normal/assembly.bam"
+    log: CFG["logs"]["gridss"] + "{seq_type}--{genome_build}/{tumour_id}--None--no_normal/gridss.log"
+    params:
+        opts = CFG["options"]["gridss"], 
+        steps = "assemble,call", 
+        jvmheap = lambda wildcards, resources: int(resources.mem_mb * 0.8)
+    conda:
+        CFG["conda_envs"]["gridss"]
+    threads:
+        CFG["threads"]["gridss"]
+    resources:
+        **CFG["resources"]["gridss"]
+    priority: 1
+    group: "enormous_bam"
+    shell:
+        op.as_one_line("""
+        gridss
+        --reference {input.fasta}
+        --output {output.vcf}
+        --workingdir `dirname {output.vcf}`
+        --assembly {output.assembly}
+        --blacklist {input.blacklist}
+        --repeatmaskerbed {input.repeatmasker}
+        --threads {threads}
+        --jvmheap {params.jvmheap}m
+        --labels "{wildcards.tumour_id}"
+        --steps {params.steps}
+        {params.opts}
+        {input.tumour_bam} 
+        2>&1 | tee -a {log}
```

Comment:
My rationale was actually so that the log files would be in the format that the developers use for their logging in case any bugs arose, but since the software is already redirecting stderr to a separate log file I will change this.